### ISSUE TITLE
Proteus 2.2 crashes Rack 2.1.2 on MacOS 13.0 M1 Mini

### ISSUE THREAD

**Issue by fractalgee** [abluenautilus/SeasideModularVCV#2]
Proteus 2.2 crashes Rack 2.1.2 quite readily after reopen of a patch that I had running for hours. It crashes either sometimes after open of patch or if for example right-clicking module. This is after save and reopen, and if newMelody() is triggered while it is opening at startup (in autosave) then it will crash on start. Here are the two variants of the stack on this M1 Mini (16GB) running MacOS 13.0 (Ventura) 

This is the stack when it crashes at start when by accident newMelody() is triggered while it is asked to step() first time after start:

```
Thread 5 Crashed:: com.apple.audio.IOThread.client
0   <translation info unavailable>	       0x1045672c0 ???
1   plugin.dylib                  	        0x157b77d3 void std::__1::__tree<std::__1::__value_type<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, int>, std::__1::__map_value_compare<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::__value_type<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, int>, std::__1::less<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >, true>, std::__1::allocator<std::__1::__value_type<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, int> > >::__assign_multi<std::__1::__tree_const_iterator<std::__1::__value_type<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, int>, std::__1::__tree_node<std::__1::__value_type<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, int>, void*>*, long> >(std::__1::__tree_const_iterator<std::__1::__value_type<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, int>, std::__1::__tree_node<std::__1::__value_type<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, int>, void*>*, long>, std::__1::__tree_const_iterator<std::__1::__value_type<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, int>, std::__1::__tree_node<std::__1::__value_type<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, int>, void*>*, long>) + 499
2   libsystem_platform.dylib      	    0x7ff813a35c33 _sigtramp + 51
3   plugin.dylib                  	        0x157b391c Proteus::newMelody() + 764
4   plugin.dylib                  	        0x157b85db Proteus::doStep() + 587
5   plugin.dylib                  	        0x157b4ea9 Proteus::process(rack::engine::Module::ProcessArgs const&) + 3145
6   libRack.dylib                 	         0x4f51ecf rack::engine::Module::doProcess(rack::engine::Module::ProcessArgs const&) + 143
7   libRack.dylib                 	         0x4f4aa53 rack::engine::Engine::stepBlock(int) + 1699
8   libRack.dylib                 	         0x4e78835 rack::audio::Device::processBuffer(float const*, int, float*, int, int) + 325
9   libRack.dylib                 	         0x4eaefff rack::RtAudioDevice::rtAudioCallback(void*, void*, unsigned int, double, unsigned int, void*) + 143
10  libRack.dylib                 	         0x539d90f RtApiCore::callbackEvent(unsigned int, AudioBufferList const*, AudioBufferList const*) + 335
11  libRack.dylib                 	         0x539c905 callbackHandler(unsigned int, AudioTimeStamp const*, AudioBufferList const*, AudioTimeStamp const*, AudioBufferList*, AudioTimeStamp const*, void*) + 21
12  CoreAudio                     	    0x7ff815b2d227 HALC_ProxyIOContext::IOWorkLoop() + 7213
13  CoreAudio                     	    0x7ff815b2b029 invocation function for block in HALC_ProxyIOContext::HALC_ProxyIOContext(unsigned int, unsigned int) + 63
14  CoreAudio                     	    0x7ff815cb2630 HALB_IOThread::Entry(void*) + 72
15  libsystem_pthread.dylib       	    0x7ff813a0b259 _pthread_start + 125
16  libsystem_pthread.dylib       	    0x7ff813a06c7b thread_start + 15
```

This is the variant if I right-click module after opening patch by File->Open menu item:

```
Thread 9 Crashed:: com.apple.audio.IOThread.client
0   <translation info unavailable>	       0x1009b7730 ???
1   plugin.dylib                  	        0x11c10c62 std::__1::__tree_iterator<std::__1::__value_type<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, int>, std::__1::__tree_node<std::__1::__value_type<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, int>, void*>*, long> std::__1::__tree<std::__1::__value_type<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, int>, std::__1::__map_value_compare<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::__value_type<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, int>, std::__1::less<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >, true>, std::__1::allocator<std::__1::__value_type<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, int> > >::__emplace_hint_unique_key_args<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const, int> const&>(std::__1::__tree_const_iterator<std::__1::__value_type<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, int>, std::__1::__tree_node<std::__1::__value_type<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, int>, void*>*, long>, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const, int> const&) + 130
2   libsystem_platform.dylib      	    0x7ff813a35c33 _sigtramp + 51
3   plugin.dylib                  	        0x11c104d3 Proteus::doStep() + 323
4   plugin.dylib                  	        0x11c0cea9 Proteus::process(rack::engine::Module::ProcessArgs const&) + 3145
5   libRack.dylib                 	         0x13a9ecf rack::engine::Module::doProcess(rack::engine::Module::ProcessArgs const&) + 143
6   libRack.dylib                 	         0x13a2a53 rack::engine::Engine::stepBlock(int) + 1699
7   libRack.dylib                 	         0x12d0835 rack::audio::Device::processBuffer(float const*, int, float*, int, int) + 325
8   libRack.dylib                 	         0x1306fff rack::RtAudioDevice::rtAudioCallback(void*, void*, unsigned int, double, unsigned int, void*) + 143
9   libRack.dylib                 	         0x17f590f RtApiCore::callbackEvent(unsigned int, AudioBufferList const*, AudioBufferList const*) + 335
10  libRack.dylib                 	         0x17f4905 callbackHandler(unsigned int, AudioTimeStamp const*, AudioBufferList const*, AudioTimeStamp const*, AudioBufferList*, AudioTimeStamp const*, void*) + 21
11  CoreAudio                     	    0x7ff815b2d227 HALC_ProxyIOContext::IOWorkLoop() + 7213
12  CoreAudio                     	    0x7ff815b2b029 invocation function for block in HALC_ProxyIOContext::HALC_ProxyIOContext(unsigned int, unsigned int) + 63
13  CoreAudio                     	    0x7ff815cb2630 HALB_IOThread::Entry(void*) + 72
14  libsystem_pthread.dylib       	    0x7ff813a0b259 _pthread_start + 125
15  libsystem_pthread.dylib       	    0x7ff813a06c7b thread_start + 15
```

**Comment by abluenautilus** [comment]
Thank you. Any chance you can share your autosave/patch.json file from when this crash happens?

**Comment by fractalgee** [comment]
Here you go. patch.json from autosave which is likely to crash on start or a couple of seconds after, before anything is fully drawn on screen. 

Also attached patch itself. I can get it to crash even by loading it and then right-clicking or twiddling a knob, and immediately or a few seconds later Rack will crash.

remove .zip suffix from patch file, git doesn't like .vcv files
[patch.json.zip](https://github.com/abluenautilus/SeasideModularVCV/files/9935784/patch.json.zip)
[FG_R2_132_Proteus_Fun.vcv.zip](https://github.com/abluenautilus/SeasideModularVCV/files/9935785/FG_R2_132_Proteus_Fun.vcv.zip)

Hope you find the iffy! if you need anything else, let me know.

**Comment by FMTolstoevsky** [comment]
Same story with almost instant crashes (after clicking Run in Clocked) on Win7 oldie.

**Comment by abluenautilus** [comment]
Thanks to you both! Much appreciated.

**Comment by meltdream** [comment]
Hi there, just wanted to report that I too experience crashes 8 out of 10 times when opening any saved patch that includes Proteus, running VCV Rack 2.1.2 on Win11. 
Luckily sometimes it manages to open it so that I can remove the module or the patches would become effectively un-openable unless editing it out with a text editor, but with the new save file format this has become a bit harder. 
Thanks for your work, awesome module anyway! :)

**Comment by abluenautilus** [comment]
If any of you know how to install the plugin from the github version, would appreciate to know if the latest commit fixes the problem. I do think I've fixed it! Thanks.

**Comment by fractalgee** [comment]
just built it and tested it for adding to patch (fine now) and reopening it from autosave (fine as well now) no exhaustive testing done, but given that these actions caused a crash within seconds, this build was still running after 5 minutes, so seems indeed fixed. Close this, as I see you have already submitted it to the library, but would be good to indicate to the team that this is a big bad crashing bug you fixed in 2.2.1 ;)

**Comment by abluenautilus** [comment]
phew. thanks again.
